src/arithmetic.jl

```diff
@@ -632,24 +674,22 @@ Calling prune or grade before exp may help to find the best algorithm for the ex
     :(@inbounds $prodexpr)
 end
 
+function inv0(x)
+    y = inv(x)
+    ifelse(iszero(x), zero(y), y)
+end
 
 function exp_trig(mv::MultiVector)
-    if iszero(mv)
-        return one(mv)
-    else
-        x = sqrt(sum(coefficients(mv) .^ 2))
-        return cos(x) + sin(x) * mv * inv(x)
-    end
```

Comment:
This branch might look like a nice optimization, but it introduces a type instability.

Reply:
So it does more harm than good.